block/sync.go

```diff
@@ -75,7 +70,14 @@ func (m *Manager) SettlementSyncLoop(ctx context.Context) error {
 
 				err = m.ApplyBatchFromSL(settlementBatch.Batch)
 				if err != nil {
-					return fmt.Errorf("process next DA batch. err:%w", err)
+					m.logger.Error("Apply batch from SL", "err", err)
+					break
+				}
+
+				// if height havent been updated, we are stuck
+				// this covers the scenario where no applicable blocks were found in the DA
+				if m.State.NextHeight() == currH {
```

Comment:
can u elaborate how this can happen? afaiu we got into the loop because we have new settlement height, which means the blocks are in the da. assuming no fraud (as it's handled by blocks unavaliable in the DA), when will this case happen?